src/Store/AuthContext/AuthReducer.ts

```diff
@@ -0,0 +1,160 @@
+import {State,Action,UserData,SigninPages,ChangePassword,SigninUser,SignedInUserInfo} from './AuthContext.types'
+import {Dispatch,SetStateAction} from 'react'
+import {ResponseTemplate} from '../../Generics.types'
+// import axios from 'axios'
+import axios from '../../useAxios'
+import {successToast,warningToast,infoToast} from '../../Components/'
+
+export const authReducer=(state:State,action:Action)=>{
```

Comment:
You can give a proper name to the Action type(like AuthAction)....as you gave for QuizAction for consistency.